### PR TITLE
Convert ZipFileProcessor to a component

### PR DESCRIPTION
**Pull request by doncem**
### Change description ###

There's no easy way to split commits nor change involved in this PR. Portion perhaps can be extracted separately but sooner or later - big bang of componentisation and inclusion into constructs will happen.

This PR helps out for future task to come in and do a re-upload stale envelopes under `CREATED` state (or even `UPLOAD_FAILURE` can go in :woman_shrugging: )

Will try to sum up:

- Moving signature configuration items to `ZipFileProcessor` where it is actually needed
- Along the way moving actions made by `ZipVerifiers` leaving former component just process the zip and any `ZipFileProcessor` consumers will not need to worry about passing functions of some sorts
- Removing not needed constructs which were used in integration tests
- Extracted tests related to signature separately as they are config based

No new functionality is introduced - just shifting around existing implementation

**Does this PR introduce a breaking change?** (check one with "x")

```
[x] Yes // i hope not
[x] No // should be not
```

### CODE REVIEW

**Comment by doncem** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 17
- Added 5
           

Complexity increasing per file
==============================
- src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTestForSignatureUsage.java  1
         

Clones added
============
- src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTest.java  1
- src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTestForSignatureUsage.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/bulk-scan-processor/pullRequest?prid=5327987&bid=17439485)

**Comment by doncem** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 17
- Added 5
           

Complexity increasing per file
==============================
- src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTestForSignatureUsage.java  1
         

Clones added
============
- src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTestForSignatureUsage.java  1
- src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/bulk-scan-processor/pullRequest?prid=5327987&bid=17439485)

**Comment by doncem** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 17
- Added 5
           

Complexity increasing per file
==============================
- src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTestForSignatureUsage.java  1
         

Clones added
============
- src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTest.java  1
- src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTestForSignatureUsage.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/bulk-scan-processor/pullRequest?prid=5327987&bid=17439485)

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/ProcessorTestSuite.java [hunk 1]
```diff
@@ -126,22 +126,6 @@ public void setUp(Construct<T> processorConstruct) throws Exception {
...
-            envelopeProcessor,
-            envelopeRepository,
-            processEventRepository,
-            containerMappings,
-            ocrValidator,
-            serviceBusHelper,
-            SIGNATURE_ALGORITHM,
-            DEFAULT_PUBLIC_KEY_BASE64,
-            paymentsEnabled
-        );
```
Comment (rkondratowicz) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/ProcessorTestSuite.java:141]: 🎊

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/processor/FailedDocUploadProcessor.java [hunk 2]
```diff
@@ -35,36 +32,19 @@
...
-        DocumentProcessor documentProcessor,
-        EnvelopeProcessor envelopeProcessor,
-        EnvelopeRepository envelopeRepository,
-        ProcessEventRepository eventRepository,
-        ContainerMappings containerMappings, // NOSONAR
-        OcrValidator ocrValidator, // NOSONAR
-        ServiceBusHelper serviceBusHelper, // NOSONAR
-        String signatureAlg,
-        String publicKeyDerFilename,
-        boolean paymentsEnabled // NOSONAR
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/processor/FailedDocUploadProcessor.java:63]: 🎉

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java [hunk 3]
```diff
@@ -94,49 +95,22 @@ public BlobProcessorTask(
...
         EnvelopeProcessor envelopeProcessor,
+        ZipFileProcessor zipFileProcessor,
         EnvelopeRepository envelopeRepository,
         ProcessEventRepository eventRepository,
         ContainerMappings containerMappings,
         OcrValidator ocrValidator,
         @Qualifier("notifications-helper") ServiceBusHelper notificationsQueueHelper,
-        @Value("${process-payments.enabled}") boolean paymentsEnabled) {
+        @Value("${process-payments.enabled}") boolean paymentsEnabled
+    ) {
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java:105]: 👍

**Comment by doncem** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 17
- Added 5
           

Complexity increasing per file
==============================
- src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTestForSignatureUsage.java  1
         

Clones added
============
- src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTest.java  1
- src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTestForSignatureUsage.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/bulk-scan-processor/pullRequest?prid=5327987&bid=17439485)

**Pull request merged**